apps/src/codebridge/InfoPanel/index.tsx

```diff
@@ -0,0 +1,99 @@
+import React, {useState} from 'react';
+
+import Button from '@cdo/apps/componentLibrary/button';
+import InstructionsView from '@cdo/apps/lab2/views/components/Instructions';
+import PanelContainer from '@cdo/apps/lab2/views/components/PanelContainer';
+import {useAppSelector} from '@cdo/apps/util/reduxHooks';
+
+import HelpAndTips from './HelpAndTips';
+
+import moduleStyles from './styles/info-panel.module.scss';
+
+enum Panels {
+  Instructions = 'Instructions',
+  HelpAndTips = 'Help and Tips',
+}
+
+export const InfoPanel = React.memo(() => {
+  const mapReference = useAppSelector(
+    state => state.lab.levelProperties?.mapReference
+  );
+  const referenceLinks = useAppSelector(
+    state => state.lab.levelProperties?.referenceLinks
+  );
+  const [currentPanel, setCurrentPanel] = useState(Panels.Instructions);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  const renderHeaderButton = () => {
+    return panelOptions().length > 1 ? (
+      <div>
+        <Button
+          icon={{
+            iconStyle: 'solid',
+            iconName: isDropdownOpen ? 'caret-up' : 'caret-down',
+          }}
+          isIconOnly
+          onClick={() => setIsDropdownOpen(!isDropdownOpen)}
+          color={'black'}
+          ariaLabel={'Information panel dropdown'}
+          size={'xs'}
+        />
+      </div>
+    ) : null;
+  };
+
+  const panelOptions = () => {
+    // For now, always include Instructions panel.
+    // TODO: support hiding this panel completely if there are no instructions.
+    const panelOptions = [Panels.Instructions];
+    if (mapReference || referenceLinks) {
+      panelOptions.push(Panels.HelpAndTips);
+    }
+    return panelOptions;
+  };
+
+  const showCurrentPanel = () => {
+    switch (currentPanel) {
+      case Panels.Instructions:
+        return <InstructionsView />;
+      case Panels.HelpAndTips:
+        return <HelpAndTips />;
+    }
+  };
```

Comment:
In your Start Over PR, you used `useCallback` for functions like `resetToStartSource` and `onClickStartOver`. Could we do the same thing here to avoid re-creating this function unless the dependencies change? 

I have the same question for `changePanel` and `renderHeaderButton` in this file.

Reply:
I'm pretty sure this can be a statically wired dispatch lookup instead.

Move it outside the component and name as:
```
const getCurrentPanel = {
  [Panels.Instructions] : InstructionsView,
  [Panels.HelpAndTips] : HelpAndTips
}
```

Then call it as:
```
{ getCurrentPanel[currentPanel]() }
```

Or, to make it prettier, but less compact:
```
const CurrentPanel = getCurrentPanel[currentPanel]
...
<CurrentPanel/>
```

Or, if you still want it inline:
```
const CurrentPanel = useMemo(() => {
  const panelLookup = {
    [Panels.Instructions] : InstructionsView,
    [Panels.HelpAndTips] : HelpAndTips
  }
  return panelLookup[currentPanel]
}, [currentPanel]);
...
<CurrentPanel/>
```

Reply:
I'll update this to be a static lookup.
To the question on `useCallback`, the reason I added it around the start over logic was because I was seeing some issues with unexpected resets of the project code due to us firing a callback too often. However, I probably was overly cautious and didn't clean up ones that didn't need to be wrapped in useCallback.